contracts/SponsorFaucet.sol

```diff
@@ -3,90 +3,62 @@ pragma experimental ABIEncoderV2;
-
-    // ------------------------------------------------------------------------
-    // Remove commission privilege for address `user` from some contract.
-    // ------------------------------------------------------------------------
-    function remove_privilege(address[] memory) public {
-    }
-}
+import "./InternalContract.sol";
 
 contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard {
+    using SafeMath for uint256;
     struct detail {
         uint256 cnt;
         uint256 limit;
     }
 
     uint256 public gas_bound;
-    uint256 public storage_bound;
+    uint256 public collateral_bound;
     mapping(address=>detail) public dapps;
-    mapping(address=>uint256) public sponsors;
     
-    event applied(address dapp);
-    event receieved(address sponsor, uint256 amount);
-
+    event applied(address indexed dapp);
+    
     SponsorWhitelistControl cpc = SponsorWhitelistControl(0x0888000000000000000000000000000000000001);
     
-    constructor(uint256 bound, uint256 s_bound) public {
-        gas_bound = bound;
-        storage_bound = s_bound;
+    constructor(uint256 gasBound, uint256 collateralBound) public {
+        gas_bound = gasBound;
+        collateral_bound = collateralBound;
     }
     
     /*** Dapp dev calls ***/ 
     //apply cfx for gas & storage
-    function applyFor(address dapp) public onlyOwner nonReentrant whenNotPaused {
-        cpc.set_sponsor_for_gas.value(gas_bound)(dapp, gas_bound);
-        cpc.set_sponsor_for_collateral.value(storage_bound)(dapp);
-        dapps[dapp].cnt++;
-        dapps[dapp].limit += (gas_bound+storage_bound);
+    function applyFor(address dapp) public nonReentrant whenNotPaused {
+        cpc.set_sponsor_for_gas.value(gas_bound)(dapp, gas_bound.div(1000));
+        cpc.set_sponsor_for_collateral.value(collateral_bound)(dapp);
+        dapps[dapp].cnt.add(1);
+        dapps[dapp].limit.add(gas_bound.add(collateral_bound));
         emit applied(dapp);
     }
 
     //accept sponsor's cfx
     function () external payable {
-        sponsors[msg.sender] += msg.value;
     }
 
     //get dapp sponsored balance
     // todo: add internal contract getMethod
-    function getDappBalance(address dapp) public view returns(uint256){
-
+    function getDappGas(address dapp) public view returns(uint256){
+        
     }
-    
-    //balance of sponsor
-    function getBalance() public view returns(uint256) {
-        return address(this).balance;
+
+    function getDappCollateral(address dapp) public view returns(uint256){
+
     }
 
-    //withdraw to specific address
-    function withdraw(address payable sponsor) public onlyOwner nonReentrant whenPaused {
-        require(address(this).balance > 1);
-        require(sponsor.send(address(this).balance-1));
+    //withdraw to specific address by amount
+    function withdraw(address payable sponsor, uint256 amount) public onlyOwner nonReentrant whenPaused {
+        require(address(this).balance > amount, "amount too high");
```

Comment:
why not support to withdraw all balance?

Reply:
fixed